app/src/main/java/org/mozilla/focus/urlinput/UrlInputFragment.kt

```diff
@@ -188,11 +188,14 @@ class UrlInputFragment : Fragment(), UrlInputContract.View, View.OnClickListener
     }
 
     private fun onCommit() {
-        val input = urlView.autocompleteResult.formattedText.let {
-            when {
-                it.isEmpty() || !URLUtil.isValidUrl(urlView.text.toString()) -> urlView.text.toString()
-                else -> it
+        val input = urlView.autocompleteResult?.let { result ->
+            if (result.text.isEmpty() || !URLUtil.isValidUrl(urlView.text.toString())) {
+                urlView.text.toString()
+            } else {
+                result.text
             }
+        } ?: run {
```

Comment:
run() seems unnecessary here?

Reply:
If `urlView.autocompleteResult == null`, will run this statement.